extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTable.java

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.engine.table.Table;
+import org.apache.iceberg.Snapshot;
+import org.jetbrains.annotations.NotNull;
+
+public interface IcebergTable extends Table {
```

Comment:
I think we might want to note that these methods are optional, and in practice only supported when the implementation requested supports manual updating.